### PR TITLE
Various version upgrades and remove jcentral

### PR DESCRIPTION
**Pull request by rockets3600**
- Simple Flat mapper alternative
- log4j upgrade
- Java 17 and Gradle 7.2 upgrade
- remove jcentral

### CODE REVIEW

**Comment by chokoswitch** [comment]
Thanks @rockets3600 looks like there's something wrong with docker hub with the CI will look into that and this PR over next few days

**Comment by rockets3600** [comment]
> Thanks @rockets3600 looks like there's something wrong with docker hub with the CI will look into that and this PR over next few days

Thank you!

**Comment by chokoswitch** [comment]
@rockets3600 I think there may be something up with the base branch of this PR, GitHub is showing "This branch has conflicts that must be resolved" though not actually showing any conflicting files which is weird. Could you check the branch is OK so that message goes away?

**Comment by rockets3600** [comment]
@chokoswitch Should be okay now.

File: common/server/framework/src/main/java/org/curioswitch/common/server/framework/database/smf/AsmInstantiatorDefinitionFactory.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
+package org.curioswitch.common.server.framework.database.smf;
```
Comment (chokoswitch) [inline common/server/framework/src/main/java/org/curioswitch/common/server/framework/database/smf/AsmInstantiatorDefinitionFactory.java:1]: @rockets3600 Can you run `./gradlew spotlessApply`?

**Comment by dsypniewski** [comment]
@chokoswitch 
> looks like there's something wrong with docker hub with the CI

I think it might be missing images on docker hub for `curiostack/java-cloud-runner`, there are no tags there at all but curiostack expects those images to be there.

**Comment by chokoswitch** [comment]
@rockets3600 Is the check for allowing edits by maintainers set on the PR by any chance? Was trying to push some tweaks to this PR but seem to not have the permission to do so.

**Comment by rockets3600** [comment]
@chokoswitch I was looking for the option to "Allow edits by maintainers", and couldn't find it. I looks like there's a limitation in github which prevents the option when the fork is under an organization instead of an individual. Could you add comments to the PR and we'll update it?

https://github.com/github/feedback/discussions/5634

**Comment by chokoswitch** [comment]
@rockets3600 Oh I see - in that case, for now can you just pull in upstream and repush? If there are still changes needed after that I'll probably create a separate PR with these same commits

**Comment by rockets3600** [comment]
Okay, thank you!

**Comment by chokoswitch** [comment]
Thanks @rockets3600 - I added one commit and merged in a separate PR

https://github.com/curioswitch/curiostack/pull/633/commits/8c2d6809e8501ff09bf782cc5e59616c140b2f24

**Comment by chokoswitch** [comment]
@rockets3600 Thanks for the patience, these changes have been released as part of the 0.9.1 release

https://github.com/curioswitch/curiostack/releases/tag/curiostack-0.9.1

**Comment by rockets3600** [comment]
@chokoswitch Huge thanks! Much appreciated.

**Pull request closed without merging**